### DISCUSSION TITLE
Clean up obsolete verified files after rename of test class, method, or parameters

### DISCUSSION THREAD

**Discussion by egil** [VerifyTests/Verify#531] [Q&A]
I am still getting comfortable with using Verify and the related workflow. One thing that does annoy me is when I refactor my tests, e.g. change their name, add parameters, or rename types, all verified files gets stranded. If its a straight up refactor, the content of the old and new verified files should be the same, so GIT can usually detect this as a rename. 

However, if I have many tests in the same folder, it can be a bit tedious to go find the verified files that are no longer relevant and delete those. Is there a way for Verify to automatically detect these scenarios and do the clean up for me? Or is there another pattern I should be using, e.g. store verified files in subfolders matching the name of the test?

This is a bigger issue when using `theory`/data driven tests, which I do quite a bit, where there can be many verified files per test.

**Comment by SimonCropp** [comment]
i would really like to fix this. it annoys me as well. 

My current way of handling it is: delete all verified files, run all the tests, do a bulk accept of all changes. This is clunky.

The problem is i havnt figured out a way to achieve this. Any ideas?

**Reply by egil** [reply]
In theory, Verify should know all the verified files that should exist after a full test run. That means it can also tell what files that should not, by looking through the test project folder and finding all verified files that do exist.

But I guess the problem Verify only knows about individual tests in isolation.

So, a few ideas:

- a custom test logger that verify can use to get info about all tests on a project level. Combine that with a `dotnet verify` tool that will run all tests in a project with the custom logger and then do the clean up after.
- with xUnit it's possible to create custom "test frameworks" that allows for complete control over test discovery, execution, etc.. That should give you all info you need. But that might not be as easy with nunit, etc., and thus not a path you want to go down.

That's all I have ATM.
